server/plugin/webhook.go

```diff
@@ -945,8 +945,8 @@ func (p *Plugin) postPullRequestReviewCommentEvent(event *github.PullRequestRevi
 		post.AddProp(postPropGithubObjectType, githubObjectTypePRReviewComment)
 
 		post.ChannelId = sub.ChannelID
-		if err = p.client.Post.CreatePost(post); err != nil {
-			p.client.Log.Warn("Error webhook post", "post", post, "error", err.Error())
+		if _, appErr := p.API.CreatePost(post); appErr != nil {
+			p.client.Log.Warn("Error webhook post", "post", post, "error", appErr.Error())
```

Comment:
`p.API` should not be used. The returned `model.AppError` makes things way too hard. `p.client` is they way to go.

Reply:
@hanzei Please refer to this comment: https://github.com/mattermost/mattermost-plugin-github/pull/768#discussion_r1649367568

Reply:
In this case, the `p.client.Post` method has essentially a bug in it, that has caused issues in this plugin because it mutates the post that is passed in. Is it really a "bug"? Maybe not, but I think we should avoid using it since it has caused issues in production that we then need to fix

2/5 The benefits of using `error` over `AppError` in this case are not worth it imo. I think keeping it as `p.client.Post` will make potential issues come back, just as they did in this PR

Reply:
@Kshitij-Katiyar I'm thinking we should keep the changes in this PR to a minimum so let's keep this as-is